### PR TITLE
Code formatting

### LINKED ISSUE #32: Use code formatter
As discussed in #3 we should use a formatter (black) to ensure consistent code style

### PR DESCRIPTION
**Pull request by mschmidt87**
This PR formats all python files in the library, tests, and examples using the black formatter and flake8.

The line length is set to 99 characters.
Flake8 is configured to ignore three rules:
- E203: This is currently broken and complains about whitespaces in list indices although that is encouraged by PEP8.  https://www.flake8rules.com/rules/E203.html (https://github.com/psf/black/issues/315)
- W503: This is outdated. https://www.flake8rules.com/rules/W503.html
- E402: Currently deactivated, because we have the manual import of the package in the test files which will be removed in a separate PR (see ticket: #26 ). https://www.flake8rules.com/rules/E402.html

The PR also adds the checks to CI so that Travis automatically checks for this in future.

Note: You can use pre-commit hooks to make your life easier. Just add the config file like this to your local repo (I could also add this to this PR? @jakobj ):
```
repos:
-   repo: https://github.com/ambv/black
    rev: 19.3b0
    hooks:
    - id: black
      language_version: python3.8
      args: [--config=pyproject.toml]
-   repo: https://github.com/pre-commit/pre-commit-hooks
    rev: v1.2.3
    hooks:
    - id: flake8
      args: [--config=.flake8]
```
Here are the steps to install the pre-commit hooks:

    Install pre-commit: pip install pre-commit
    Define .pre-commit-config.yaml with the hooks you want to include. (see above)
    Execute pre-commit install to install git hooks in your .git/ directory.

Note: This PR is branched off from #34 , so #34 should be merged first.

Fixes #32

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[flake8]
+ignore = E203, W503, E402
+max-line-length = 99
+max-complexity = 18
+select = B,C,E,F,W,T4,B9
+exclude = __init__.py
```
Comment (jakobj) [inline .flake8]: why exclude `__init__.py`?

**Comment by jakobj** [comment]
regarding pre-commit hooks: looks really cool, but i don't think we need the file in the repository, do we? it seems boiler plate black and flake8

File: .flake8 (hunk 1 above)
Comment (mschmidt87) [reply .flake8]: > regarding pre-commit hooks: looks really cool, but i don't think we need the file in the repository, do we? it seems boiler plate black and flake8

Yes, agree.

Comment (mschmidt87) [reply .flake8]: I guess because flake8 then always complains about unused imports.

**Comment by mschmidt87** [comment]
> regarding pre-commit hooks: looks really cool, but i don't think we need the file in the repository, do we? it seems boiler plate black and flake8

Okay.

**Comment by jakobj** [comment]
for some reason travis fails, i've restarted the jobs now

**Comment by mschmidt87** [comment]
Travis passes now. The construction in travis is not elegant because we first `cd` into tests and then `cd` out of it after executing the testsuite, but I'll keep it like this for now, and we can revise this.
The reason why we need to `cd` into the test folder is the construction in the test scripts where we manually add the path to the library to the system path. This will be changed soon anyway (#26 ) and so we can tackle this in a separate PR.

**Pull request merged**